library/core/src/ascii/ascii_char.rs

```diff
@@ -449,6 +453,7 @@ impl AsciiChar {
     /// or returns `None` if it's too large.
     #[unstable(feature = "ascii_char", issue = "110998")]
     #[inline]
+    #[ensures(|result| (b <= 127) == result.is_some())]
```

Comment:
Can we strengthen this to ensure that the wrapped value is correct? E.g.
```suggestion
    #[ensures(|result| (b <= 127) == (result.is_some() && result.unwrap() as u8 == b)]
```
?

Reply:
Wouldn't this panic in the case where `result.is_none()`?

Reply:
If `result` is `None` then the RHS would short circuit to false (because `result.is_some()` is false) and the `unwrap` won't be called.

Reply:
Ah, right! I hadn't realized the new parenthesis (before `result.is_some()`) so I was reading `result.unwrap() as u8 == b` as a separate condition.

Reply:
Addressed as part of a separate commit 👍